bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/connectivity/rawdata/LocalConnectRawDataBean.java

```diff
@@ -0,0 +1,252 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solax.internal.connectivity.rawdata;
+
+import java.util.Arrays;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solax.internal.model.InverterData;
+import org.openhab.binding.solax.internal.model.InverterType;
+import org.openhab.binding.solax.internal.util.GsonSupplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * The {@link LocalConnectRawDataBean} collects the raw data and the specific implementation to return the parsed data.
+ * If there are differences between the inverters probably would be wise to split the parsing in seprate class(es)
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class LocalConnectRawDataBean implements RawDataBean, InverterData {
+
+    private final Logger logger = LoggerFactory.getLogger(LocalConnectRawDataBean.class);
+
+    private @Nullable String sn;
+    private @Nullable String ver;
+    private int type;
+    @SerializedName("Data")
+    private short @Nullable [] data;
+    @SerializedName("Information")
+    private String @Nullable [] information;
+    private @Nullable String rawData;
+
+    @Override
+    public String toString() {
+        return "LocalConnectRawDataBean [sn=" + sn + ", ver=" + ver + ", type=" + type + ", Information="
+                + Arrays.toString(information) + ", Data=" + Arrays.toString(data) + "]";
+    }
+
+    public @Nullable String getSn() {
+        return sn;
+    }
+
+    public void setSn(@Nullable String sn) {
+        this.sn = sn;
+    }
+
+    public @Nullable String getVer() {
+        return ver;
+    }
+
+    public void setVer(@Nullable String ver) {
+        this.ver = ver;
+    }
+
+    public int getType() {
+        return type;
+    }
+
+    public void setType(int type) {
+        this.type = type;
+    }
+
+    public short @Nullable [] getData() {
+        return data;
+    }
+
+    public void setData(short @Nullable [] data) {
+        this.data = data;
+    }
+
+    public String @Nullable [] getInformation() {
+        return information;
+    }
+
+    public void setInformation(String @Nullable [] information) {
+        this.information = information;
+    }
+
+    @Override
+    public @Nullable String getRawData() {
+        return rawData;
+    }
+
+    public void setRawData(String rawData) {
+        this.rawData = rawData;
+    }
+
+    public static LocalConnectRawDataBean fromJson(String json) {
+        if (json.isEmpty()) {
+            throw new IllegalStateException("Argument should be not null or empty.");
```

Comment:
`IllegalArgumentException` is more correct here:
```suggestion
            throw new IllegalArgumentException("JSON payload should not be empty");
```